src/descriptor/mod.rs

```diff
@@ -848,7 +978,153 @@ impl<Pk: MiniscriptKey> fmt::Display for Descriptor<Pk> {
+
+        // This would check all the consensus rules for p2sh/p2wsh and
+        // even tapscript in future
+        Ctx::check_ms_validity(&ms)?;
+
+        Ok(Self {
+            k,
+            pks: Vec::from(pks),
+            phantom: PhantomData,
+        })
+    }
+    /// Parse an expression tree into a SortedMultiVec
+    fn from_tree(tree: &Tree) -> Result<Self, Error>
+    where
+        <Pk as FromStr>::Err: ToString,
+    {
+        let k = tree.args[0].name.parse::<usize>().unwrap(); // FIXME: unwrap
+        let pks: Result<Vec<Pk>, _> = tree.args[1..]
+            .iter()
+            .map(|sub| expression::terminal(sub, Pk::from_str))
+            .collect();
+        pks.map(|pks| SortedMultiVec::new(k, &pks))?
+    }
+}
+
+impl<Pk: MiniscriptKey, Ctx: ScriptContext> SortedMultiVec<Pk, Ctx> {
+    fn translate_pk<FPk, Q, FuncError>(
+        &self,
+        translatefpk: &mut FPk,
+    ) -> Result<SortedMultiVec<Q, Ctx>, FuncError>
+    where
+        FPk: FnMut(&Pk) -> Result<Q, FuncError>,
+        Q: MiniscriptKey,
+    {
+        let pks: Result<Vec<Q>, _> = self.pks.iter().map(&mut *translatefpk).collect();
+        Ok(SortedMultiVec {
+            k: self.k,
+            pks: pks?,
+            phantom: PhantomData,
+        })
+    }
+}
+
+impl<Pk: MiniscriptKey + ToPublicKey, Ctx: ScriptContext> SortedMultiVec<Pk, Ctx> {
+    /// Create Terminal::Multi containing sorted pubkeys
+    pub fn sorted_node(&self) -> Terminal<bitcoin::PublicKey, Ctx> {
+        let mut pks: Vec<bitcoin::PublicKey> =
+            self.pks.iter().map(|pk| pk.to_public_key()).collect();
+        // Sort pubkeys lexigraphically according to BIP 67
+        pks.sort_by(|a, b| a.to_string().partial_cmp(&b.to_string()).unwrap());
+        Terminal::Multi(self.k, pks)
+    }
+
+    fn encode(&self) -> script::Script {
+        self.sorted_node()
+            .encode(script::Builder::new())
+            .into_script()
+    }
+
+    // FIXME: is this trait bound right?
+    /// Attempt to produce a satisfying witness for the
+    /// witness script represented by the parse tree
+    fn satisfy<S: Satisfier<bitcoin::PublicKey>>(&self, satisfier: S) -> Option<Vec<Vec<u8>>> {
+        match satisfy::Satisfaction::satisfy(&self.sorted_node(), &satisfier).stack {
+            satisfy::Witness::Stack(stack) => Some(stack),
+            satisfy::Witness::Unavailable => None,
+        }
+    }
+
+    fn script_size(&self) -> usize {
+        script_num_size(self.k)
+            + 1
+            + script_num_size(self.pks.len())
+            + self
+                .pks
+                .iter()
+                .map(ToPublicKey::serialized_len)
+                .sum::<usize>()
+    }
+
+    fn max_satisfaction_witness_elements(&self) -> usize {
+        2 + self.k
+    }
+
+    fn max_satisfaction_size(&self, _: usize) -> usize {
+        1 + 73 * self.k
+    }
+}
+
+impl<Pk: MiniscriptKey, Ctx: ScriptContext> crate::policy::Liftable<Pk>
```

Comment:
I'm not convinced that we should implement `lift` directly on `SortedMultiVec`, but if we do, we should definitely sort the keys before lifting.

Reply:
I don't fully understand what `lift` does so I'll defer to you guys on that.

IIRC @sanket1729 said that pubkeys didn't need to be sorted for this.

Also, I just realized that [the `lift` impl for `Descriptor` doesn't have the `ToPublicKey` trait bound](https://github.com/rust-bitcoin/rust-miniscript/blob/master/src/policy/mod.rs#L119). Would we want to add it?

Reply:
We definitely do not want to add a `ToPublicKey` bound to the lift feature of descriptor...I think the existing code is fine actually, after talking to Sanket.